### PR TITLE
additonal features

### LINKED ISSUE #12: Parse errors from remote compilation and make them clickable

### PR DESCRIPTION
**Pull request by ghostbuster91**
This closes #12 #82

### CODE REVIEW

File: src/main/kotlin/com/elpassion/intelijidea/MFStateProvider.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    override fun loadState(state: State) {
+        isTurnOn = state.isTurnOn
+    }
+
+    override fun getState(): State = myState
+
+    private val myState = State()
+
+    class State {
```
Comment (marcinsus) [inline src/main/kotlin/com/elpassion/intelijidea/MFStateProvider.kt]: Duplicate with src/main/kotlin/com/elpassion/intelijidea/common/MFStateProvider.kt

File: src/main/kotlin/com/elpassion/intelijidea/common/console/RemoteToLocalTranslatingInputFilter.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+class RemoteToLocalTranslatingInputFilter(private val project: Project) : InputFilter {
+
+    private val pattern = Pattern.compile("$FILE_REGEX:$LINE_REGEX: $ERROR_REGEX", Pattern.MULTILINE)
+
+    override fun applyFilter(text: String, contentType: ConsoleViewContentType) = mutableListOf(Pair(translateText(text), contentType))
+
+    private fun translateText(text: String): String {
+        val matcher = pattern.matcher(text)
+        return if (!matcher.find()) {
```
Comment (marcinsus) [inline src/main/kotlin/com/elpassion/intelijidea/common/console/RemoteToLocalTranslatingInputFilter.kt]: 'When' will be shorter and more readable.

File: src/main/kotlin/com/elpassion/intelijidea/common/console/RemoteToLocalTranslatingInputFilter.kt [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            text
+        } else {
+            text.replace("\\D+mainframer/${project.name}".toRegex(), project.basePath!!)
```
Comment (marcinsus) [inline src/main/kotlin/com/elpassion/intelijidea/common/console/RemoteToLocalTranslatingInputFilter.kt]: If you extract one pattern to regex then i suggest do it also with other.

File: src/main/kotlin/com/elpassion/intelijidea/MFStateProvider.kt (hunk 1 above)
Comment (ghostbuster91) [inline src/main/kotlin/com/elpassion/intelijidea/MFStateProvider.kt]: Invalid. It was moved into common package in other commit.

File: src/main/kotlin/com/elpassion/intelijidea/common/console/RemoteToLocalTranslatingInputFilter.kt (hunk 2 above)
Comment (ghostbuster91) [inline src/main/kotlin/com/elpassion/intelijidea/common/console/RemoteToLocalTranslatingInputFilter.kt]: I'm not sure about when to boolean application.

File: src/main/kotlin/com/elpassion/intelijidea/MFStateProvider.kt (hunk 1 above)
Comment (ghostbuster91) [inline src/main/kotlin/com/elpassion/intelijidea/MFStateProvider.kt]: Done

**Pull request merged**